src/Settings.pm

```diff
@@ -428,13 +428,16 @@ sub loadByHandle {
 	assert(defined $object) if DEBUG;
 
 	my $filename;
-	if ($object->{autoSearch}) {
-		if ($object->{type} == CONTROL_FILE_TYPE) {
-			$filename = _findFileFromFolders($object->{name}, \@controlFolders);
-		} else {
-			$filename = _findFileFromFolders($object->{name}, \@tablesFolders);
-		}
-	} else {
+	my $internalFilename = $object->{internalName} || $object->{name};
+
+	#hooks of type 'pre_load_' make it possible to change the filename before openkore searches for it (this filename must contain file name and file path)
+	my %pre_load = (internalFilename => $internalFilename, filename => \$filename);
```

Comment:
I'm not a big fan of passing scalar refs around as in/out variables. Perl doesn't make them easy to work with; when using them, one has to use the clunky double sigil `${$pre_load->{filename}}` to read and write them. While it takes an extra line here to extract the modified `$filename`, it becomes easier for any hook handlers to work with the data.

OpenKore does use this idiom elsewhere though. Whatever you prefer is fine. :)